data_store/controllers/ingest.py

```diff
@@ -496,14 +497,15 @@ def save_submission_file_s3(excel_file: FileStorage, submission_id: str):
         .with_entities(Submission.id, Fund.fund_code, Programme.programme_name)
         .distinct()
     ).one()
-
     upload_file(
         file=excel_file,
         bucket=Config.AWS_S3_BUCKET_SUCCESSFUL_FILES,
         object_name=f"{fund_type}/{str(uuid)}",
         metadata={
             "submission_id": submission_id,
-            "filename": f"{fund_type}-{str(uuid)}.xlsx",
+            "filename": make_ascii_safe_filename(
+                excel_file.filename or f"{fund_type}-{programme_name}-{submission_id}.xlsx"
```

Comment:
Will this not still throw an error as the excel filename will always have some value and there is no try-catch? (sorry if I missed something)

Reply:
It should always be defined during the file upload process. I’ve included a fallback in case the filename is empty, None, or if something goes wrong during ascii validation. This is also to  a warning from mypy.